### DISCUSSION TITLE
Tuyamcu isn't communicating at all, I'm not sure what to try next

### DISCUSSION THREAD

**Discussion by TheJoe1990** [arendst/Tasmota#16534] [Q&A]
Tywe1s on a tyjw2 v2.0.0, flashed with raspberry pi and soldered leads. Module type is set to tuyamcu (54). Have tried pine 1,3 and 13,15 and both baud rates. Still getting no communication from the device. 

I didn't back up the original firmware because I hadn't foreseen this outcome and I'm not sure where to go from here. 

[Posting the log in the next message]

![20220914_183940](https://user-images.githubusercontent.com/36345748/190475629-2315cb79-744b-487e-8f7d-d2a264355381.jpg)

**Comment by TheJoe1990** [comment]
`18:44:23.409 TYA: Send "55aa00000000ff"
18:44:34.424 TYA: Send "55aa00000000ff"
18:44:39.425 TYA: Send "55aa001c00080116090f122c2704bb"
18:44:40.360 WIF: Checking connection...
18:44:41.423 WIF: Sending Gratuitous ARP
18:44:45.415 TYA: Send "55aa00000000ff"
18:44:54.406 RSL: STATE = {"Time":"2022-09-15T18:44:54","Uptime":"0T08:05:14","UptimeSec":29114,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER":"ON","Wifi":{"AP":1,"SSId":"Home Automations","BSSId":"D6:21:F9:65:3E:56","Channel":6,"Mode":"11n","RSSI":96,"Signal":-52,"LinkCount":1,"Downtime":"0T00:00:09"}}
18:44:56.438 TYA: Send "55aa00000000ff"
18:45:01.125 WIF: Checking connection...
18:45:07.401 TYA: Send "55aa00000000ff"
18:45:18.416 TYA: Send "55aa00000000ff"
18:45:21.089 WIF: Checking connection...
18:45:29.413 TYA: Send "55aa00000000ff"
18:45:39.445 TYA: Send "55aa001c00080116090f122d2704bc"
18:45:40.435 TYA: Send "55aa00000000ff"
18:45:41.357 WIF: Checking connection...
18:45:41.407 WIF: Sending Gratuitous ARP
18:45:51.441 TYA: Send "55aa00000000ff"
18:46:01.346 WIF: Checking connection...
18:46:02.440 TYA: Send "55aa00000000ff"
18:46:13.449 TYA: Send "55aa00000000ff"
18:46:21.861 WIF: Checking connection...
18:46:24.425 TYA: Send "55aa00000000ff"
18:46:35.448 TYA: Send "55aa00000000ff"
18:46:39.448 TYA: Send "55aa001c00080116090f122e2704bd"
18:46:41.436 WIF: Sending Gratuitous ARP
18:46:42.360 WIF: Checking connection...
18:46:46.410 TYA: Send "55aa00000000ff"
18:46:57.413 TYA: Send "55aa00000000ff"
18:47:02.673 WIF: Checking connection...
18:47:08.430 TYA: Send "55aa00000000ff"
18:47:19.423 TYA: Send "55aa00000000ff"
18:47:23.089 WIF: Checking connection...
18:47:30.433 TYA: Send "55aa00000000ff"
18:47:39.430 TYA: Send "55aa001c00080116090f122f2704be"
18:47:41.427 WIF: Sending Gratuitous ARP
18:47:41.438 TYA: Send "55aa00000000ff"
18:47:43.835 WIF: Checking connection...
18:47:52.428 TYA: Send "55aa00000000ff"
18:48:03.434 TYA: Send "55aa00000000ff"
18:48:03.877 WIF: Checking connection...
18:48:14.410 TYA: Send "55aa00000000ff"
18:48:24.092 WIF: Checking connection...
18:48:25.399 TYA: Send "55aa00000000ff"
18:48:36.427 TYA: Send "55aa00000000ff"
18:48:39.431 TYA: Send "55aa001c00080116090f12302704bf"
18:48:41.422 WIF: Sending Gratuitous ARP
18:48:44.354 WIF: Checking connection...
18:48:47.407 TYA: Send "55aa00000000ff"
18:48:58.429 TYA: Send "55aa00000000ff"
18:49:04.868 WIF: Checking connection...
18:49:09.434 TYA: Send "55aa00000000ff"
18:49:20.425 TYA: Send "55aa00000000ff"
18:49:25.335 WIF: Checking connection...
18:49:31.438 TYA: Send "55aa00000000ff"
18:49:39.454 TYA: Send "55aa001c00080116090f12312704c0"
18:49:41.417 WIF: Sending Gratuitous ARP
18:49:42.425 TYA: Send "55aa00000000ff"
18:49:45.838 WIF: Checking connection...
18:49:53.447 TYA: Send "55aa00000000ff"
18:49:54.403 RSL: STATE = {"Time":"2022-09-15T18:49:54","Uptime":"0T08:10:14","UptimeSec":29414,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER":"ON","Wifi":{"AP":1,"SSId":"Home Automations","BSSId":"D6:21:F9:65:3E:56","Channel":6,"Mode":"11n","RSSI":92,"Signal":-54,"LinkCount":1,"Downtime":"0T00:00:09"}}
18:50:04.441 TYA: Send "55aa00000000ff"
18:50:05.868 WIF: Checking connection...
18:50:15.400 TYA: Send "55aa00000000ff"
18:50:26.416 TYA: Send "55aa00000000ff"
18:50:26.608 WIF: Checking connection...
18:50:37.415 TYA: Send "55aa00000000ff"
18:50:39.414 TYA: Send "55aa001c00080116090f12322704c1"
18:50:41.417 WIF: Sending Gratuitous ARP
18:50:47.092 WIF: Checking connection...
18:50:48.416 TYA: Send "55aa00000000ff"
18:50:59.410 TYA: Send "55aa00000000ff"
18:51:07.358 WIF: Checking connection...
18:51:10.403 TYA: Send "55aa00000000ff"
18:51:21.449 TYA: Send "55aa00000000ff"
18:51:27.366 WIF: Checking connection...
18:51:32.431 TYA: Send "55aa00000000ff"
18:51:33.028 HTP: Console
18:51:39.041 HTP: Console
18:51:39.421 TYA: Send "55aa001c00080116090f12332704c2"`

**Comment by TheJoe1990** [comment]
After trying to brute force the solution to this, setting all pins to RX except one at a time as TX and trying both BAUD rates on all configs, ended up over 90 reboots, still no joy.  Decided to try ESPHome, with pins 1 and 3, 13 and 15, both baud rates, still no joy.  Not sure what to do next.  

Ive tried to fallow the steps in this link: https://github.com/arendst/Tasmota/discussions/10470

I dont understand why we switch to:
GPIO13 - SerBr Rx
GPIO15 - SerBr Tx

**Comment by TheJoe1990** [comment]
During further testing in ESPHome I attempted to bridge wires for TX and RX going to the MCU, this verified we are on pins GPIO 13 and 15, tested at both baud rates, and still not getting nothing.  Reached out to the seller to try and get another wifi module.  Open to any ideas you have.

**Comment by dlasher** [comment]
Having same issues, any closer to a solution?

**Reply by TheJoe1990** [reply]
Hopefully you backed up the chip before you flashed because no, I never did get this working and I would still like to.

The most help I got was on the ESPhome discord. The next step that was suggested was to log all the incoming and outgoing signals with factory firmware, then post that on the ESPhome discord so they could help me further.

I'm planning to revisit this at some point. If you find a solution though please let me know. =]

**Comment by barbudor** [comment]
At no point you have documented what is the device you are trying to hack
Are you sure that there is a TuyaMCU ?

TYJW2 is just an ESP8266 module with a serial connector.
Up to you to analyse that serial connection with
- an oscilloscope
- a logic analyser
- a PC connected with a USB/serial adapter

**Comment by PaulG4H** [comment]
https://github.com/htmltiger/tcl-electriq-split-ac/tree/main
